### ISSUE TITLE
TS Server fatal error:  Cannot read property 'indexOf' of undefined

### ISSUE THREAD

**Issue by RatTac** [microsoft/vscode#116330]
Issue Type: <b>Bug</b>

❗️❗️❗️ Please fill in the sections below to help us diagnose the issue ❗️❗️❗️

**TypeScript Version:** 4.1.4

**Steps to reproduce crash**
1.
2.
3.

**TS Server Log**
❗️Server logging disabled. To help us fix crashes like this, please enable logging by setting:
```json
"typescript.tsserver.log": "verbose"
```
After enabling this setting, future crash reports will include the server log.

**TS Server Error Stack**
Server: `semantic`
```
TypeError: Cannot read property 'indexOf' of undefined
    at Object.parsePackageName (c:\Users\ZOTNOELL\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:39472:30)
    at ConfiguredProject.Project.enablePlugin (c:\Users\ZOTNOELL\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:151962:24)
    at ConfiguredProject.enablePluginsWithOptions (c:\Users\ZOTNOELL\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:152477:30)
    at ProjectService.loadConfiguredProject (c:\Users\ZOTNOELL\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:154206:25)
    at ProjectService.createAndLoadConfiguredProject (c:\Users\ZOTNOELL\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:154145:22)
    at ProjectService.createLoadAndUpdateConfiguredProject (c:\Users\ZOTNOELL\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:154150:36)
    at ProjectService.assignProjectToOpenedScriptInfo (c:\Users\ZOTNOELL\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:155055:44)
    at c:\Users\ZOTNOELL\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:155395:97
    at Object.flatMap (c:\Users\ZOTNOELL\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:513:25)
    at ProjectService.applyChangesInOpenFiles (c:\Users\ZOTNOELL\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:155395:41)
    at Session.handlers.ts.Map.ts.getEntries._a.<computed> (c:\Users\ZOTNOELL\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:156413:46)
    at c:\Users\ZOTNOELL\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:158286:88
    at IOSession.Session.executeWithRequestId (c:\Users\ZOTNOELL\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:158277:28)
    at IOSession.Session.executeCommand (c:\Users\ZOTNOELL\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:158286:33)
    at IOSession.Session.onMessage (c:\Users\ZOTNOELL\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:158310:35)
    at Interface.<anonymous> (c:\Users\ZOTNOELL\AppData\Local\Programs\Microsoft VS Code\resources\app\extensions\node_modules\typescript\lib\tsserver.js:160520:27)
    at Interface.emit (events.js:315:20)
    at Interface._onLine (readline.js:329:10)
    at Interface._normalWrite (readline.js:474:12)
    at Socket.ondata (readline.js:186:10)
    at Socket.emit (events.js:315:20)
    at addChunk (_stream_readable.js:295:12)
    at readableAddChunk (_stream_readable.js:271:9)
    at Socket.Readable.push (_stream_readable.js:212:10)
    at Pipe.onStreamRead (internal/stream_base_commons.js:186:23)
```

VS Code version: Code 1.53.1 (5d424b828ada08e1eb9f95d6cb41120234ef57c7, 2021-02-08T23:29:42.785Z)
OS version: Windows_NT x64 10.0.16299

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Xeon(R) CPU E5-2680 v4 @ 2.40GHz (28 x 2394)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: enabled<br>opengl: enabled_on<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: enabled_on<br>video_decode: enabled<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|undefined|
|Memory (System)|63.88GB (51.86GB free)|
|Process Argv|--crash-reporter-id b5406e75-0e86-4f9f-a260-9f0e3db4224d|
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (6)</summary>

Extension|Author (truncated)|Version
---|---|---
simple-react-snippets|bur|1.2.3
bracket-pair-colorizer-2|Coe|0.2.0
vscode-eslint|dba|2.1.14
gitlens|eam|11.2.1
prettier-vscode|esb|5.9.1
autoimport|ste|1.5.3

</details>

**Comment by RatTac** [comment]
In my case I could fix it within the tsconfig.json:

I had a section

"plugins": [
			{
				"name": "typescript-is",
				"transform": "typescript-is/lib/transform-inline/transformer"
			},
			{
				"name": "ts-transformer-keys",
				"transform": "ts-transformer-keys/transformer"
			}
		]

Before it did not have the name keys. After adding the name keys everything seems to work as expected...

**Comment by gjsjohnmurray** [comment]
/duplicate of #116219 and will be fixed in upcoming 1.53.2
